buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/gradlebuild/unittestandcompile/UnitTestAndCompilePlugin.kt

```diff
@@ -139,11 +139,14 @@ class UnitTestAndCompilePlugin : Plugin<Project> {
     fun Project.addGeneratedResources(gradlebuildJava: UnitTestAndCompileExtension) {
         val classpathManifest = tasks.register("classpathManifest", ClasspathManifest::class)
         java.sourceSets["main"].output.dir(mapOf("builtBy" to classpathManifest), gradlebuildJava.generatedResourcesDir)
-        plugins.withType<IdeaPlugin> {
-            configure<IdeaModel> {
-                module {
-                    resourceDirs = resourceDirs + gradlebuildJava.generatedResourcesDir
-                    testResourceDirs = testResourceDirs + gradlebuildJava.generatedTestResourcesDir
+        val isExecutingIdeaTask = gradle.startParameter.taskNames.contains("idea")
+        if (isExecutingIdeaTask) {
+            plugins.withType<IdeaPlugin> {
+                configure<IdeaModel> {
+                    module {
+                        resourceDirs = resourceDirs + gradlebuildJava.generatedResourcesDir
```

Comment:
My expectation is that removing this declaration would break running integration tests when using the native import without module per source set. Does this declaration have any effect in the module per source set kind of import? If not, we can leave it in until we switched.

Reply:
Yes you are right. That breaks. I didn't know that we actually use that already.

Adding the resource dir has an impact on the 'module per source set' import. It adds the additional resource folder to the corresponding main module as another content root (as the folder is then outside the module). This _shouldn't hurt_ but I don't like it very much as it is not necessary and might still lead to unnecessary problems.

I checked if we can change the condition such that it behave different based on the 'module per source set' setting, but I think it is not possible to determine that in the build script.

So my assumption was wrong. Merging this PR would break something.

I really would like to put this condition in when we start using the 'module per source set'. Maybe we just wait with the merge until the new EAP is released this week?

Or I remove the condition and we add it later.... But I don't like it very much to maintain **three** different IDE import methods in parallel and juggling all these specifics.

Reply:
I reverted the change. I will add a note to gradle/gradle-private#1675 to remove this workaround completely later.